operator/standalone/pipeline.go

```diff
@@ -23,27 +23,29 @@ func fetchHelmRelease(ctx context.Context) error {
 	return err
 }
 
-// fetchOperatorConfig fetches a matching v1alpha1.PostgresqlStandaloneOperatorConfig from the OperatorNamespace.
+// fetchOperatorConfigF fetches a matching v1alpha1.PostgresqlStandaloneOperatorConfig from the OperatorNamespace.
 // The Major version specified in v1alpha1.PostgresqlStandalone is used to filter the correct config by the v1alpha1.PostgresqlMajorVersionLabelKey label.
 // If there is none or multiple found, it returns an error.
-func fetchOperatorConfig(ctx context.Context) error {
-	list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
-	labelMatch := client.MatchingLabels{
-		v1alpha1.PostgresqlMajorVersionLabelKey: getInstanceFromContext(ctx).Spec.Parameters.MajorVersion.String(),
-	}
-	ns := client.InNamespace(getOperatorNamespaceFromContext(ctx))
-	err := getClientFromContext(ctx).List(ctx, list, labelMatch, ns)
-	if err != nil {
-		return err
-	}
-	if len(list.Items) == 0 {
-		return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labelMatch, ns)
-	}
-	if len(list.Items) > 1 {
-		return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labelMatch, ns)
+func fetchOperatorConfigF(operatorNamespace string) func(ctx2 context.Context) error {
```

Comment:
Why we have renamed this function? Also the argument variable name `ctx2` doesn't look great

Reply:
because the func returns another func, and not directly executing stuff anymore. So I thought it's worth to make it clear that upon invoking fetchOperatorConfigF(), nothing gets executed yet. But yeah, I fixed the `ctx2`